### PR TITLE
chore: create build artifacts for each commit

### PR DESCRIPTION
**Pull request by DevVersion**
**ToDo's**
- [x] Need a GitHub token for `travis_after_all` and for pushing the builds
- [x] Need a repository like `material-builds`

Wrote a plugin to wait for all Travis modes to finish ([travis-after-modes](https://github.com/DevVersion/travis-after-modes))

### CODE REVIEW

**Comment by DevVersion** [comment]
@jelbourn Should be good for review. Now using the personal repository as discussed for testing.

File: scripts/ci/after-success.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
```
Comment (jelbourn) [inline scripts/ci/after-success.sh]: Add a top-level comment explaining the purpose of this file

File: scripts/release/publish-build-artifacts.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
```
Comment (jelbourn) [inline scripts/release/publish-build-artifacts.sh]: Add a top-level comment explaining the purpose of this file

File: scripts/release/publish-build-artifacts.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Go to the project root directory
+cd $(dirname $0)/../..
+
+BUILD_DIR="dist/@angular/material"
+BUILD_VERSION=`sed -nE 's/^\s*"version": "(.*?)",$/\1/p' package.json`
```
Comment (jelbourn) [inline scripts/release/publish-build-artifacts.sh]: Prefer using `$( ... )` over ticks for bash:
```sh
# Extract the current version from package.json
BUILD_VERSION=$(sed -nE 's/^\s*"version": "(.*?)",$/\1/p' package.json)
```
(here and elsewhere)

**Pull request merged**